NYPLCardCreator/BusinessLogic/FlowCoordinator.swift

```diff
@@ -0,0 +1,144 @@
+//
+//  FlowCoordinator.swift
+//  NYPLCardCreator
+//
+//  Created by Ernest Fan on 2021-11-04.
+//  Copyright © 2021 NYPL Labs. All rights reserved.
+//
+
+import UIKit
+
+public class FlowCoordinator {
+  internal let urlSession: URLSession
+  public let configuration: CardCreatorConfiguration
+  internal var authToken: ISSOToken?
+  
+  public init(configuration: CardCreatorConfiguration) {
+    self.configuration = configuration
+    self.urlSession = URLSession(configuration: .ephemeral)
+  }
+  
+  public func startRegularFlow(completion: @escaping (Result<UINavigationController>) -> Void) {    
+    authenticate(using: configuration.platformAPIInfo) { [weak self] result in
+      guard let self = self else {
+        return
+      }
+
+      switch result {
+      case .success(let authToken):
+        self.authToken = authToken
+        OperationQueue.main.addOperation {
+          let introVC = IntroductionViewController(configuration: self.configuration, authToken: authToken)
+          completion(.success(UINavigationController(rootViewController: introVC)))
+        }
+      case .fail(let error):
+        OperationQueue.main.addOperation {
+          completion(.fail(FlowCoordinator.errorWithUserFriendlyMessage(amending: error)))
+        }
+      }
+    }
+  }
+  
+  // MARK: - Fetch ISSOToken
```

Comment:
Everything below this line is moved from FlowCoordinator+Juvenile with very minor changes